core/arch/arm32/sta/sub.mk

```diff
@@ -1,2 +1,7 @@
 srcs-y += sta_self_tests.c
 srcs-y += core_self_tests.c
+
+ifeq ($(CFG_SE_API),y)
+srcs-${CFG_SE_API_SELF_TEST} += se_api_self_tests.c
```

Comment:
Okay. Got your point. I will remove it. The dependency should be enforced by a configuration system.